src/contexts/SiteLaunchContext.tsx

```diff
@@ -39,7 +41,10 @@ export const SiteLaunchProvider = ({
   initialStepNumber,
 }: SiteLaunchProviderProps): JSX.Element => {
   const { siteName } = useParams<{ siteName: string }>()
-
+  const [
+    isSiteLaunchBlockedToastShown,
+    setIsSiteLaunchBlockedToastShown,
+  ] = useState<boolean>(false)
```

Comment:
could i get some context on why we need this? seems like it's to avoid multiple re-renders?

Reply:
yes plus when the redirection originally occurs, that happens from another page (`siteLaunchPad`). This is to store the state between these two pages.

Reply:
Can we rebase on top of the toast PR to avoid having to do this in the context?

Reply:
still need this bah to sync state between the two pages! is there a better way to do that without prop drilling?

Reply:
to be clear ya, this is already rebased on top of the error toast pr already

Reply:
seems like the main usage is 
```ts
if (!shouldUseSiteLaunchFeature(siteName)) {
    setIsSiteLaunchBlockedToastShown(true)
  }
``` in `SiteLaunchPad` - because this set state is given by a function call, it should not matter whether the function call happens in site launch pad or site dashboard. 

would be more in favour of 
```ts
useEffect(() => {
    if (!shouldUseSiteLaunchFeature(siteName)) {
      errorToast({
        id: "no_access_to_launchpad",
        description: "You do not have access to this page.",
      })
    }
  }, [
    errorToast,
    siteName
  ])
```

which makes the flow clearer and allows us to bypass all the set state + prop passing. the key thing here is to decide what's computed data and what's actually state. in this case, the decision to show the toast appears to be computed data so we should just use the computation directly. (also can hide the computation itself in context if you want)

Reply:
hmm wait ah wouldnt this then re-render everytime the site dashboard is rendered?

Reply:
why would it re-render? the `errorToast` dep already existed prior + `siteName` is constant for a given site, no?

Reply:
resolved offline - put the `useEffect` in `SiteLaunchPad` works without having to go through this hassle